### PR TITLE
Connector Updates

### LINKED ISSUE #1389: Premature close mysql when building source
## Steps To Reproduce

Steps we can take to reproduce the issue you're seeing. Sample data is helpful.

### Environment

- Node version (`v21.4.0`):
- npm version (`10.2.4`):
- OS: Docker
- Browser: Chrome
- Database: PostgreSQL
- Packages version (updating to the newest on 2023-12-06):
```
    "@evidence-dev/bigquery": "usql",
    "@evidence-dev/core-components": "usql",
    "@evidence-dev/csv": "usql",
    "@evidence-dev/evidence": "usql",
    "@evidence-dev/faker-datasource": "usql",
    "@evidence-dev/mysql": "usql",
    "@evidence-dev/postgres": "usql",
    "@evidence-dev/sqlite": "usql",
    "@evidence-dev/usql-datatable": "1.0.2",
```

## Expected Behavior

Build sources without error, or at least fail build - now it's silently skips that sources.

## Actual Behaviour

I'm getting `Premature close` error when running `evidence sources`
```
# npm run build:sources

> XXX@0.0.1 build:sources
> evidence sources --debug

(node:109) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Processing prd_mysql_XXX
  XXX1 ✖ Premature close
  XXX2 ✖ Premature close
  XXX3 ✖ Premature close
  XXX4 ✖ Premature close
  XXX5 ✖ Parquet error: Incorrect number of rows, expected 16 != 12 rows
  XXX6 ✖ Premature close
  XXX7 ✖ Premature close
  XXX8 ✖ Premature close
  XXX9 ✖ Parquet error: Incorrect number of rows, expected 10 != 8 rows
```

## Workarounds
 
Nope

### LINKED ISSUE #1452: MSSQL Connector Prematurely Returning Failed Connection
See: https://evidencedev.slack.com/archives/C023LRB9Z40/p1704583842306609

### LINKED ISSUE #1473: Add SSL no-verify option to Postgres Connector
See #1448

### PR DESCRIPTION
**Pull request by ItsMeBrianD**
### Description

This removes legacy environment variable configuration and attempts to add manual connection closing in connectors that support it.

I could not test many of these connectors locally, so some scrutiny would be much appreciated

- Resolves #1389 
- Resolves #1473 
- Resolves #1452 

### Checklist

- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

### CODE REVIEW

**Comment by mcrascal** [comment]
ErrorChart also uses the missing /invalid creds error messages to decide if it's missing credentials, and display a link to the settings page. [See](https://github.com/evidence-dev/evidence/blob/2a135aa9942075443f7224b588aa949321056722/packages/core-components/src/lib/unsorted/viz/core/ErrorChart.svelte#L26)

File: packages/postgres/index.cjs [hunk 1]
```diff
@@ -371,6 +291,14 @@ module.exports.options = {
 						{ value: 'verify-ca', label: 'Verify CA' },
 						{ value: 'verify-full', label: 'Verify Full' }
 					]
+				},
+				rejectUnauthorized: {
```
[the remark this answers is not in the document]
Comment (ItsMeBrianD) [reply packages/postgres/index.cjs:295]: image
Per `node-postgres`

image
Per node docs

This behavior is also accomplished with `NODE_REJECT_UNAUTHORIZED=false` as an environment variable

**Comment by ItsMeBrianD** [comment]
> ErrorChart also uses the missing /invalid creds error messages to decide if it's missing credentials, and display a link to the settings page. [See](https://github.com/evidence-dev/evidence/blob/2a135aa9942075443f7224b588aa949321056722/packages/core-components/src/lib/unsorted/viz/core/ErrorChart.svelte#L26)

I don't think that error message will ever appear, because those errors will either come from components (e.g. empty dataset) or the QueryStore (e.g. missing table, invalid syntax, etc)

Source-level error handling has all been moved up to the `run sources` step / source hmr (which fires toasts)

I do think I need to do another pass on this to make sure that we are handling more cases to prevent the `Invalid Credentials` message from appearing (probably want to rethrow an error with the original as the cause if it isn't an error directly)

**Review by Winterhart — COMMENTED** [review]
LGTM, mostly config

You need help for testing a specific database connection?

File: packages/bigquery/test/test.js [hunk 2]
```diff
@@ -138,7 +148,11 @@ test('numeric types are retrieved correctly', async () => {
 test('query batches results properly', async () => {
 	const { rows, expectedRowCount } = await runQuery(
 		'select 1 union all select 2 union all select 3 union all select 4 union all select 5',
-		undefined,
+		{
+			project_id: process.env.BIGQUERY_PROJECT_ID,
+			client_email: process.env.BIGQUERY_CLIENT_EMAIL,
+			private_key: process.env.BIGQUERY_PRIVATE_KEY
```
Comment (csjh) [inline packages/bigquery/test/test.js:154]: Probably worth putting this object and the other 2 into a `const creds = { ... }`

File: packages/mssql/test/test.js [hunk 3]
```diff
@@ -60,7 +69,15 @@ test('query batches results properly', async () => {
...
 				'select 1 as one union all select 2 as two union all select 3 as three union all select 4 as four union all select 5 as five',
-				undefined,
+				{
+					user: process.env.MSSQL_USER,
+					host: process.env.MSSQL_HOST,
+					database: process.env.MSSQL_DATABASE,
+					password: process.env.MSSQL_PASSWORD,
+					port: process.env.MSSQL_PORT,
+					trust_server_certificate: process.env.MSSQL_TRUST_SERVER_CERTIFICATE,
+					encrypt: process.env.MSSQL_ENCRYPT
```
Comment (csjh) [inline packages/mssql/test/test.js:79]: Same as above

File: packages/mysql/test/test.js [hunk 4]
```diff
@@ -59,7 +67,14 @@ test('query batches results properly and predicts rows', async () => {
...
 			const { rows, expectedRowCount } = await runQuery(
 				'select 1 union all select 2 union all select 3 union all select 4 union all select 5',
-				undefined,
+				{
+					user: process.env.MYSQL_USER,
+					host: process.env.MYSQL_HOST,
+					database: process.env.MYSQL_DATABASE,
+					password: process.env.MYSQL_PASSWORD,
+					port: process.env.MYSQL_PORT,
+					ssk: process.env.MYSQL_SSL
```
Comment (csjh) [inline packages/mysql/test/test.js:76]:
```suggestion
					ssl: process.env.MYSQL_SSL
```
also same as above

**Review by csjh — APPROVED** [review]
just nitpicks, lgtm otherwise

File: packages/bigquery/index.cjs [hunk 5]
```diff
@@ -79,42 +47,44 @@ const standardizeRow = (row) => {
...
 		};
 	}
 };
 
+/**
+ *
+ * @param {BigQueryOptions} db
+ * @returns {BigQuery}
+ */
+const getConnection = (credentials) => new BigQuery({ ...credentials, maxRetries: 10 });
```
Comment (kamilglod) [inline packages/bigquery/index.cjs:82]: missing `getCredentials` call, please see https://evidencedev.slack.com/archives/C023LRB9Z40/p1705406074518099

**Pull request merged**